### DISCUSSION TITLE
Jump to first file

### DISCUSSION THREAD

**Discussion by limaceous-bushwhacker** [gokcehan/lf#1398] [Q&A] [answered]
I was wondering if there was a builtin way of jumping to the first file in a directory filled with both files and directories? Thanks in advance :)

**Comment by joelim-work** [comment, marked as answer]
There's no builtin command for this kind of movement, but assuming you have directories sorted above files this is a hacky solution:

```
map gf :set dironly true; bottom; set dironly false; down
```

Another alternative that is more flexible is to use the shell to determine the first file, and then call `lf -remote` to `select` it:

```shell
map gf %{{
    # replace this with any other method of getting a file
    file=$(ls -F | grep -v '[*/=>@|]$' | head -n 1)

    if [ -f "$file" ]; then
        lf -remote "send $id select \"$file\""
    fi
}}
```

**Comment by limaceous-bushwhacker** [comment]
Thank @joelim-work. This is a good alternative
